### PR TITLE
DOC: add NEP29 to template

### PR DESCRIPTION
**Pull request by tacaswell**

### CODE REVIEW

File: {{ cookiecutter.repo_name }}/docs/source/min_versions.rst [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  versions of Python.
+- support minor versions of ``numpy`` initially released in the 24
+  months prior to a planned project release date or the oldest
+  version that supports the minimum Python version (whichever is
+  higher).
+- The project will always support at least the 3 latest minor
+  versions of NumPy.
+
+The minimum supported version of Python will be set to
+``python_requires`` in ``setup``.  All supported minor versions of
```
Comment (mrakitin) [inline {{ cookiecutter.repo_name }}/docs/source/min_versions.rst]:
```suggestion
``python_requires`` in ``setup.py``.  All supported minor versions of
```

File: {{ cookiecutter.repo_name }}/docs/source/min_versions.rst [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+Python will be in the test matrix and have binary artifacts built
+for releases.
+
+The project should adjust upward the minimum Python and NumPy
+version support on every minor and major release, but never on a
+patch release.
+
+This is consistent with NumPy NEP 29
```
Comment (mrakitin) [inline {{ cookiecutter.repo_name }}/docs/source/min_versions.rst]: Do we have a link for this NEP? Would be much more convenient to have it as a URL.

Comment (mrakitin) [reply {{ cookiecutter.repo_name }}/docs/source/min_versions.rst]: Oh, found: https://github.com/numpy/numpy/pull/14086/files.

File: {{ cookiecutter.repo_name }}/docs/source/min_versions.rst (hunk 1 above)
Comment (tacaswell) [reply {{ cookiecutter.repo_name }}/docs/source/min_versions.rst]: This was intentionally `setup` as the function _in_ `setup.py`.  The is copy-pasted from the NEP we should keep it as-is here and move discussion of the wording to numpy.

File: {{ cookiecutter.repo_name }}/docs/source/min_versions.rst (hunk 2 above)
Comment (tacaswell) [reply {{ cookiecutter.repo_name }}/docs/source/min_versions.rst]: It does not seem to be published yet.

Comment (danielballan) [reply {{ cookiecutter.repo_name }}/docs/source/min_versions.rst]: Let's hold off on merging until it is published and link-able.

File: {{ cookiecutter.repo_name }}/docs/source/min_versions.rst [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+Minimum Version of Python and NumPy
+===================================
+
+
+- This project supports at least the minor versions of Python
+  initially released 42 months prior to a planned project release
+  date.
+- The project will always support at least the 2 latest minor
+  versions of Python.
+- support minor versions of ``numpy`` initially released in the 24
```
Comment (danielballan) [inline {{ cookiecutter.repo_name }}/docs/source/min_versions.rst]: Add "The project will" for consistency with bullet points above and below.

**Pull request merged**